### PR TITLE
The Lord of the Evals

### PR DESCRIPTION
**Pull request by tech6hutch**
Three Evals for the Elven-kings under the sky,
Seven for the Dwarf-lords in their halls of stone,
Nine for Mortal Men doomed to die,
One for the Pull Requester on his dark fork
In the Land of Komada where the Shadows lie.
One Eval to rule them all, One Eval to find them,
One Eval to bring them all and in the darkness bind them
In the Land of Komada where the Shadows lie.

Note: I haven't actually tested the Komada version yet, but here's my quick port.

### Proposed Semver Increment Bump: [MAJOR/MINOR/PATCH]
MINOR

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

* Know the exact type of what you evaled
  * Has a little notice if it's a boolean, number, or string instantiated with `new` (see [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#Distinction_between_string_primitives_and_String_objects))
  * Notes when a promise has been `await`ed, and what its resolved value is
* Know how long the eval took to execute
  * Unit may be seconds, milliseconds, or microseconds, depending on the scale of the duration, for enhanced readability
* New flags for deleting your command message, logging the output to console instead of sending it to the channel, and doing the command silently without output
* If the output is long (7+ lines, or having long lines that will wrap), it asks before posting it to the channel
* If the output is too long to send (>2000 chars), it asks if you want to log it to the console or upload it as a file

### (If Applicable) What Issue does it fix?

 Fixes... 
* The eval command silently erring if the output is more than 2000 characters

### CODE REVIEW

File: src/commands/System/eval.js [hunk 1]
```diff
@@ -1,15 +1,160 @@
+const now = require("performance-now");
+const { MessageAttachment } = require("discord.js");
 const { inspect } = require("util");
 
-/* eslint-disable no-eval */
-exports.run = async (client, msg, [code]) => {
+// The depth to inspect the evaled output to, if it's not a string
+const inspectionDepth = 0;
```
Comment (CyberiumShadow) [inline src/commands/System/eval.js]: Make these into clientOptions tbh.

`client.config.eval.inspectionDepth` etc

Comment (CyberiumShadow) [inline src/commands/System/eval.js]: Make these into a clientConfig option

`client.config.eval.inspectionDepth` etc

File: src/commands/System/eval.js [hunk 2]
```diff
@@ -1,15 +1,160 @@
...
+// const getTypeStr shouldn't recurse more than once, but just in case
+const typeRecursionLimit = 2;
```
Comment (CyberiumShadow) [inline src/commands/System/eval.js]: and here

File: src/commands/System/eval.js [hunk 3]
```diff
@@ -1,15 +1,160 @@
...
+// The number of lines before the output is considered overly long
+const tooManyLines = 7;
```
Comment (CyberiumShadow) [inline src/commands/System/eval.js]: and here

File: src/commands/System/eval.js [hunk 4]
```diff
@@ -1,15 +1,160 @@
...
+// The approx. number of chars per line in a codeblock on Android, on a Google Pixel XL
+const mobileCharsPerLine = 34;
```
Comment (CyberiumShadow) [inline src/commands/System/eval.js]: and here

**Review by CyberiumShadow — CHANGES_REQUESTED** [review]

**Comment by CyberiumShadow** [comment]
DIscussed in staff chat. Preferable to move this to the https://github.com/dirigeants/komada-pieces repo

**Comment by tech6hutch** [comment]
👌 gonna commit a simpler version, like I did on Klasa, and PR this version to Komada Pieces

**Pull request merged**